src/libImaging/TiffDecode.c

```diff
@@ -720,7 +720,11 @@ ImagingLibTiffDecode(
     }
 
  decode_err:
-    TIFFClose(tiff);
+    if (clientstate->fp) {
```

Comment:
I think a comment here explaining why we're checking this would be good.

Reply:
Ok, I've added some comments.